src/Orleans/IDs/UniqueKey.cs

```diff
@@ -59,7 +60,10 @@ public bool IsSystemGrainKey
 
         public bool HasKeyExt
         {
-            get { return IdCategory == Category.KeyExtGrain; }
+            get {
+                var category = IdCategory;
+                return category == Category.KeyExtGrain || category == Category.GeoClient;
```

Comment:
>return category == Category.KeyExtGrain || category == Category.GeoClient; [](start = 16, length = 74)

A comment that GeoClient uses key extension for passing cluster ID would help understand the ||.